### DISCUSSION TITLE
Reporting broken links?

### DISCUSSION THREAD

**Discussion by rkazak** [smlnj/smlnj#293] [Q&A] [answered]
Hi all,

Where should we report broken links that are seen on the smlnj.org website, specifically within https://smlnj.org/doc/literature.html#tutorials the link

- [Introduction to Functional Programming](https://smlnj.org/doc/pdfs/Gordon-FP-Intro.pdf) by Michael Gordon (1996)
- [Essentials of Standard ML Modules](https://smlnj.org/doc/pdfs/Tofte-modules-tutorial-1996.pdf) by Mads Tofte (1996)
- [Four Lectures on Standard ML](https://smlnj.org/doc/pdfs/Tofte-four-lectures-1989.pdf) by Mads Tofte (1989)
- [A Gentle Introduction to ML](https://www.cs.nmsu.edu/~rth/cs/cs471/sml.html) by Andrew Cumming
- [Tips for Computer Scientists On Standard ML](https://smlnj.org/doc/pdfs/Tofte-tips-SML-2009.pdf) by Mads Tofte (2009)

This one is reported as being insecure
- [Programming in Standard ML '97: An On-line Tutorial](https://www.dcs.ed.ac.uk/lfcsreps/EXPORT/97/ECS-LFCS-97-364/index.html) by Stephen Gilmore (1997)

**Comment by dmacqueen** [comment, marked as answer]
Unfortunately, there is no repository for the sources for the smlnj.org web site, which would be (i.e. the Issues thereof) the natural place to report problems like broken links on smlnj.org.  There are two source code repositories, smlnj/smlnj and smlnj/legacy containing compiler source code for two versions of the compiler (which is unfortunate!), and one could report broken links in the corresponding Issues of either of those repositories.  Alternatively, web site problems could be reported in the Issues of the smlnj/.github repository, which is mainly used to host Wiki pages independently of the two source code repositories.
